backend/internal/models/goals.go

```diff
@@ -0,0 +1,21 @@
+package models
+
+
+
+type Goal struct {
+	DbModel
+	UserId   			uint          	`json:"user_id"`
+	GameId   			uint          	`json:"game_id"` 
+	//Game     			Game          	`gorm:"game"` TODO: When seeding is finished we can officially test this
+	TeamId   			uint          	`json:"team_id"` 
+	//Team     			Team			`gorm:"team"` TODO: When seeding is finished we can officially test this
+	Duration 			uint			`json:"duration"` // Do we potentially want to change this to time.Duration
+	Period	    		uint			`json:"period"`
+	Assist1Id			uint   	   		`json:"assist1_id"`
+	Assist2Id			uint   	   		`json:"assist2_id"`
+	PlayerDifferential	int		   		`json:"powerplay"`
```

Comment:
Is there a reason why this is an int when the rest are uint?

Reply:
Tanner thought of a solution for PlayerDifferential. I will put exactly what he said to me: 

"Hey Michael, Skyler and I had an idea for the 'power_play' field in the 'Goal' data. I believe we talked about doing a boolean to say yes this was during a power play or no this was not during a power play. From yesterday the team was asking about what would happen if a team scored when the other team had a power play. Skyler and I were thinking about different scenarios that could occur during the game:
1. One person on one team is in the penalty box for the power play,
2. One person from each team is each team's penalty box,
3. Multiple people on one team are in the penalty box,
4. Multiple people on one team are in the penalty box while the other team has one person in the penalty box.
So with these scenarios, what if we changed the type to an integer which captures the difference between the amount of players on each team currently on the ice and not in the penalty box. If the teams were even then the value would be 0. If one team has a power play they would have a value of [1,2,3,4,5, or 6] with the other team having all 6 players in the penalty box. Or if the opposing team has a power play then that team would have a value of [-6,-5-4,-3,-2, or -1].
What do you think? I know you said you pushed your code already."


So I assumed that since we need negative it should not be an unsigned int.

Reply:
Ah I see, thank you!